.github/workflows/tests.yml

```diff
@@ -494,6 +494,231 @@ jobs:
+            xz-utils \
+            zfsutils-linux
+
+          mkdir -p "$(go env GOPATH)/bin"
+          curl -sSfL https://dl.min.io/server/minio/release/linux-amd64/minio --output "$(go env GOPATH)/bin/minio"
+          chmod +x "$(go env GOPATH)/bin/minio"
+
+          # Also grab the latest minio client to maintain compatibility with the server.
+          curl -sSfL https://dl.min.io/client/mc/release/linux-amd64/mc --output "$(go env GOPATH)/bin/mc"
+          chmod +x "$(go env GOPATH)/bin/mc"
+
+      - name: Download system test dependencies
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: system-test-deps
+          merge-multiple: true
+          path: /home/runner/go/bin
+
+      - name: Set exec perms on LXD binaries
+        run: |
+          set -eux
+          ls -lR /home/runner/go/bin/
+          chmod uog+x /home/runner/go/bin/*
+
+      - name: Download docs (for the objects.inv.txt file to be available for the e2e tests)
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: documentation
+          merge-multiple: true
+          path: doc/_build
+
+      - name: Run LXD daemon
+        run: |
+          set -eux
+          echo "root:1000000:1000000000" | sudo tee /etc/subuid /etc/subgid
+          export CGO_CFLAGS="-I/home/runner/work/lxd/lxd-test/vendor/dqlite/include/"
+          export CGO_LDFLAGS="-L/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+          export LD_LIBRARY_PATH="/home/runner/work/lxd/lxd-test/vendor/dqlite/.libs/"
+          export LXD_DOCUMENTATION="/home/runner/work/lxd/lxd/doc/_build/"
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo rm -rf /var/lib/lxd
+          sudo -E PATH=${PATH} LD_LIBRARY_PATH=${LD_LIBRARY_PATH} $(go env GOPATH)/bin/lxd --group sudo &
+
+      - name: Setup LXD
+        shell: bash
+        run: |
+          set -eux
+          export PATH="/home/runner/go/bin:$PATH"
+          sudo -E LXD_DIR=/var/lib/lxd lxc storage create default zfs
+          sudo -E LXD_DIR=/var/lib/lxd lxc profile device add default root disk path=/ pool=default
+          sudo -E LXD_DIR=/var/lib/lxd lxc network create local-network
+          sudo -E LXD_DIR=/var/lib/lxd lxc profile device add default eth0 nic network=local-network
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set core.https_address "[::]:8443"
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set cluster.https_address "127.0.0.1"
+          sudo -E LXD_DIR=/var/lib/lxd lxc cluster enable local
+          sudo -E LXD_DIR=/var/lib/lxd lxc config set user.show_permissions=true
+
+      - name: Checkout LXD-UI
+        uses: actions/checkout@v4
+        with:
+          repository: 'canonical/lxd-ui'
+          ref: main
+          path: lxd-ui
+
+      - name: Install Dotrun
+        run: sudo pip3 install dotrun
+
+      - name: Restore cached keys
```

Comment:
Could we have those generated on the fly please? You can potentially reuse `test/includes/certificates.sh` that has a handy function called `gen_cert_and_key`.

Reply:
Generation of the keys is taking ~1 minute and burning cpu cycles on each ci run seems easy to resolve, hence we introduced this cache a while ago in the lxd-ui pipeline. It is just a few kb and I think the benefits outweigh here. wdyt?

Reply:
Hmm, maybe that was due to RSA being slow as with our function, this should be even faster than fetching from the cache:

```
$ . test/includes/certificates.sh
$ export LXD_CONF=$(mktemp -d)
$ time gen_cert_and_key lxd-ui
-----

real	0m0.007s
user	0m0.003s
sys	0m0.004s
```

So hard to beat :)

And you get such cert/key:

```
$ openssl x509 -noout -text -in $LXD_CONF/lxd-ui.crt
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            0b:d3:1c:7f:64:54:df:b9:34:af:1f:cc:40:14:bd:8b:b2:92:03:0f
        Signature Algorithm: ecdsa-with-SHA384
        Issuer: CN = lxd-ui.local
        Validity
            Not Before: Nov 14 23:13:44 2024 GMT
            Not After : Nov 15 23:13:44 2024 GMT
        Subject: CN = lxd-ui.local
        Subject Public Key Info:
            Public Key Algorithm: id-ecPublicKey
                Public-Key: (384 bit)
                pub:
                    04:bd:8c:4e:94:7d:bb:66:de:f7:52:a8:05:f8:19:
                    d4:32:0b:4c:8a:b2:60:42:77:27:33:7e:74:cb:af:
                    39:66:70:3c:d9:f6:79:26:c8:5c:43:cf:1e:b9:e5:
                    ee:02:42:55:ec:04:15:c8:4f:9a:93:ad:69:9c:87:
                    dd:95:c4:52:69:66:93:22:5a:e0:77:65:51:2e:fd:
                    76:21:28:87:23:82:8d:55:f6:8d:1e:2c:54:0e:1a:
                    98:1f:8f:4f:fd:44:98
                ASN1 OID: secp384r1
                NIST CURVE: P-384
        X509v3 extensions:
            X509v3 Subject Key Identifier: 
                C8:FC:A2:AC:F1:1C:85:0E:E8:7C:FD:A3:A3:7B:27:AE:D0:8C:54:BB
            X509v3 Authority Key Identifier: 
                C8:FC:A2:AC:F1:1C:85:0E:E8:7C:FD:A3:A3:7B:27:AE:D0:8C:54:BB
            X509v3 Basic Constraints: critical
                CA:TRUE
    Signature Algorithm: ecdsa-with-SHA384
    Signature Value:
        30:65:02:31:00:82:02:fc:69:6f:ec:88:f6:9f:53:94:b2:68:
        76:1c:b6:f4:33:7a:0f:62:49:df:7e:95:a7:37:b5:9a:df:78:
        08:a2:14:d4:57:4e:7e:ec:db:75:eb:7c:5d:c3:75:ac:cf:02:
        30:4f:b3:c5:22:63:a3:6b:c6:00:7c:98:80:cd:91:1e:62:11:
        5a:2c:e0:72:d0:6c:db:d7:d4:93:85:79:0c:8b:02:5f:0c:2b:
        bc:47:ab:06:b0:c3:d4:64:1d:ad:76:20:9f
```

Reply:
Oh, this is very nice. I will definitely use it instead, and we don't need the cache. Thanks for the hint :+1: